src/lib/y2packager/dialogs/addon_selector.rb

```diff
@@ -229,6 +233,38 @@ def product_description(product)
         # render the ERB template in the context of this object
         erb.result(binding)
       end
+
+      # return a list of the preselected products
+      # during upgrade we want to preselect the installed products
+      # @return [Array<Y2Packager::ProductLocation>] the products
+      def preselected_products
```

Comment:
code looks fine, but can you please find a better location for this piece of code? As I would like to use it also in autoupgrade to select addons to upgrade, so I prefer to call it from some non-dialog location.